parser/src/aspects/literal_parser.rs

```diff
@@ -133,29 +139,31 @@ impl<'a> LiteralParser<'a> for Parser<'a> {
     }
 
     fn parse_literal(&mut self) -> ParseResult<LiteralValue> {
-        let token = self.next_token()?;
+        let token = self.cursor.next()?;
         match token.token_type {
             TokenType::IntLiteral => Ok(LiteralValue::Int(token.value.parse::<i64>().map_err(
-                |e| match e.kind() {
-                    IntErrorKind::PosOverflow | IntErrorKind::NegOverflow => {
-                        self.mk_parse_error("Integer constant is too large.")
+                |e| {
+                    match e.kind() {
+                        IntErrorKind::PosOverflow | IntErrorKind::NegOverflow => self
+                            .expected::<()>("Integer constant is too large.")
+                            .unwrap_err(),
+                        _ => self.expected::<()>(&e.to_string()).unwrap_err(),
```

Comment:
Shouldn't we keep `self.mk_parse_error()` to avoid this `unwrap_err`? `ParseError` can't be initialized directly, as it may depend on the current token in the future.

Reply:
haven't focused on errors as we might rework error handling in a future PR to have clear messages on the relevent error, because current error messages are fatal (one error ends the whole parsing process) and completely unclear to the user (even to us sometimes!) i think we should make a respectable error reporting system in the future, let's discuss that in a later issue.

havn't minded of this, (+ this code comes from you!) i think we could let this as is and rethink all error reports later.